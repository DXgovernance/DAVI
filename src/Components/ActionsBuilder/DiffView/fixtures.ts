export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
