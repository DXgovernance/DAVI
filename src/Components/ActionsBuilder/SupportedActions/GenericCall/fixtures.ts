import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'Components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const genericDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.GENERIC_CALL,
  function: ERC20Contract.getFunction('increaseAllowance'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    param1: BigNumber.from(10000),
    param2: BigNumber.from(10000),
    param3: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
