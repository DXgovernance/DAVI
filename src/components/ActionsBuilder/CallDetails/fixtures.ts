import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
