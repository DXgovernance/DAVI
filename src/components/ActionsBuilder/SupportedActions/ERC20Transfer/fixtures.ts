import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
