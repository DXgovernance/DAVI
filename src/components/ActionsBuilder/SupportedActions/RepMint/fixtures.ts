import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber, utils } from 'ethers';
import ERC20SnapshotRep from 'contracts/ERC20SnapshotRep.json';

const ERC20SnapshotRepContract = new utils.Interface(ERC20SnapshotRep.abi);

export const repMintEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '',
  value: BigNumber.from(0),
  args: {
    to: '',
    amount: BigNumber.from(0),
  },
};

export const repMintDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
    amount: BigNumber.from(10000),
  },
};
