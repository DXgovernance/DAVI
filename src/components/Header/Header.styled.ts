import styled from 'styled-components';
import { Box, Container } from 'components/primitives/Layout';
import { Heading } from 'components/primitives/Typography';
import { ReactComponent as DaviLogo } from '../../assets/images/project_davi_logo_white.svg';

export const HeaderWrapper = styled.nav`
  padding: 0.75rem 0;
  max-height: 80px;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: ${({ theme }) => theme.colors.bg1};
  z-index: 200;
  @media only screen and(min-width: 768px) {
    padding: 1.5rem 0;
  }

  border-bottom: 1px solid ${({ theme }) => theme.colors.border1};
`;

export const HeaderContainer = styled(Container)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0;
  color: ${({ theme }) => theme.colors.text};
`;

export const MenuItems = styled(Box)`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-left: auto;
`;

export const ClickableHeading = styled(Heading)`
  cursor: pointer;
`;

export const TextLogo = styled(DaviLogo)`
  width: 40%;
  max-height: 50px;
`;
