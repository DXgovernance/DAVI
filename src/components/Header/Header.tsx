import { useTranslation } from 'react-i18next';
import { useNavigate } from 'react-router-dom';
import { useNetwork } from 'wagmi';
import { NetworkButton } from '../NetworkButton';
import { WalletButton } from '../WalletButton';
import {
  HeaderWrapper,
  HeaderContainer,
  MenuItems,
  ClickableHeading,
} from './Header.styled';

const Header = () => {
  const navigate = useNavigate();
  const { t } = useTranslation();
  const { chain } = useNetwork();

  return (
    <HeaderWrapper as="header">
      <HeaderContainer>
        <ClickableHeading
          onClick={() => navigate(`/${chain.network}`)}
          size={2}
        >
          <strong>{t('guilds.guilds')}</strong>
        </ClickableHeading>
        <MenuItems>
          <NetworkButton />
          <WalletButton />
        </MenuItems>
      </HeaderContainer>
    </HeaderWrapper>
  );
};

export default Header;
