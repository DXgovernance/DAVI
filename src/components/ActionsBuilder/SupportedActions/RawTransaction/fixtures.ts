import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber } from 'ethers';
import { MOCK_ADDRESS } from 'hooks/Guilds/ens/fixtures';

export const rawDataCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.RAW_TRANSACTION,
  function: null,
  to: MOCK_ADDRESS,
  value: BigNumber.from('1000000000000000000'),
  args: null,
  optionalProps: {
    data: '0x01',
  },
};
