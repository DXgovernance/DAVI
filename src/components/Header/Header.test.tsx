import Header from './Header';
import { render } from 'utils/tests';
import { mockChain } from 'components/Web3Modals/fixtures';

jest.mock('contexts/Guilds/orbis', () => ({}));

jest.mock('wagmi', () => ({
  chain: {},
  useAccount: () => ({ isConnected: false }),
  useNetwork: () => ({ chain: mockChain, chains: [mockChain] }),
  useSwitchNetwork: () => ({ switchNetwork: jest.fn() }),
  useConnect: () => ({ connect: jest.fn(), connectors: [] }),
  useDisconnect: () => ({ disconnect: jest.fn() }),
}));

jest.mock('provider/ReadOnlyConnector', () => ({
  READ_ONLY_CONNECTOR_ID: 'readOnly',
}));

jest.mock('contexts/Guilds/transactions', () => ({
  useTransactions: () => ({ transactions: [] }),
}));

describe('Header', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
