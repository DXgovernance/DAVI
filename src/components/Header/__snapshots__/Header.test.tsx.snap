// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `
.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  cursor: pointer;
  font-size: 14px;
  border: 1px solid #303338;
  background-color: #295FF4;
  color: #fff;
  border-radius: 32px;
  padding: 0.5rem 0.8rem;
  margin: 0.2rem;
  width: auto;
}

.c8:disabled {
  color: initial;
  opacity: 0.4;
  cursor: auto;
}

.c8:hover:enabled {
  border-color: #fff;
}

.c8:active:enabled {
  border: 1px solid #303338;
}

.c4 {
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 20px;
  line-height: 1.2;
  font-weight: 600;
}

.c1 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c2 {
  width: 1200px;
  max-width: calc(100% - 1.25rem);
  margin: 0 auto;
}

.c0 {
  padding: 0.75rem 0;
  max-height: 80px;
  position: -webkit-sticky;
  position: -webkit-sticky;
  position: sticky;
  top: 0;
  background: #1B1D1F;
  z-index: 200;
  border-bottom: 1px solid #303338;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0;
  color: #fff;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-left: auto;
}

.c5 {
  cursor: pointer;
}

.c6 {
  width: 40%;
  max-height: 50px;
}

@media only screen and (min-width:768px) {
  .c2 {
    padding: 3rem 0rem;
  }
}

@media only screen and(min-width:768px) {
  .c0 {
    padding: 1.5rem 0;
  }
}

<div>
   
  <header
    class="c0"
  >
    <div
      class="c1 c2 c3"
    >
      <h2
        class="c4 c5"
        data-testid="project-name"
        size="2"
      >
        <svg
          class="c6"
        >
          project_davi_logo_white.svg
        </svg>
      </h2>
      <div
        class="c1 c7"
      >
        <button
          class="c8"
        >
          notConnected
        </button>
        <button
          class="c7"
          data-testid="connect-wallet-btn"
        >
          connectWallet
        </button>
      </div>
    </div>
  </header>
   
</div>
`;
